main.tf

```
@@ -20,26 +23,37 @@ locals {
   # Gitlab Agent CI/CD variables
   gitlab_agent_kubernetes_context_variables = {
     (var.gitlab_agent_variable_name_agent_id) : gitlab_cluster_agent.this.name,
-    (var.gitlab_agent_variable_name_agent_project) : data.gitlab_project.this.path_with_namespace,
+    (var.gitlab_agent_variable_name_agent_project) : local.use_existing_project == 1 ? data.gitlab_project.this[0].path_with_namespace : gitlab_project.project[0].path_with_namespace,
```

```suggestion
    (var.gitlab_agent_variable_name_agent_project) : local.project_path_with_namespace,
```

This change is dependent from the `project_path_with_namespace` local variable.